flymake-shellcheck.el

```diff
@@ -30,13 +31,22 @@
 
 ;;; Code:
 
+(defgroup flymake-shellcheck nil
+  "Shellcheck backend for Flymake."
+  :prefix "flymake-shelcheck-"
+  :group 'tools)
+
+(defcustom flymake-shellcheck-path
+  (executable-find "shellcheck")
+  "The path to the `shellcheck' executable."
+
+  :type 'string
+  :group 'flymake-shellcheck)
+
 (defvar-local flymake-shellcheck--proc nil)
 
 (defun flymake-shellcheck--backend (report-fn &rest _args)
   "Shellcheck backend for Flymake.  Check for problems, then call REPORT-FN with results."
-  (unless (executable-find
-           "shellcheck") (error "Could not find shellcheck executable"))
```

Comment:
Maybe we could leave this check, but for `flymake-shellcheck-path` instead?

Reply:
What would you think about instead defining the default value of `flymake-shellcheck-path` as `(or executable-find "shellcheck") (error "Could not find shellcheck executable in PATH"))`?  I don't know if that would be considered idiomatic, and I'm fine with putting the check in the `defun` instead;  my thought is just that I would personally rather get an error when loading than when running.

Reply:
I think using a default value that potentially signals an error could be a bit confusing in some contexts, for example if the package is being loaded during someone's `init.el`. I believe it's better to do the check here (because there's no other place that I can think of where we could add it).

Reply:
> I think using a default value that potentially signals an error could be a bit confusing in some contexts

Sure.  That seems reaonable.

> there's no other place that I can think of where we could add it

Indeed.  I think the only other place it could _feasibly_ make sense to perform the check would be in the `load` function, but that would pose a lot of same problems as checking in `defcustom`.

I've re-added a slightly modified version of this check as 32263d8c3c7c221c1b8a944811bff4c41a769f51.

Reply:
Excellent! Only change I would add is a check for `nil`:
```elisp
(unless (and flymake-shellcheck-path (file-executable-p flymake-shellcheck-path))
    (error "Could not find shellcheck executable"))
```
...otherwise the call to `file-executable-p` could signal an error itself.

Reply:
Oof.  Good catch.  2233b4908f6567ed473a0c7889f5c5f44eeeab80 implements the fix.